File: gatling-performance-suite/src/test/scala/com/example/performance/simulations/SmokeSimulation.scala
package com.example.performance.simulations

import io.gatling.core.Predef._
import io.gatling.http.Predef._
import com.example.performance.scenarios.HomepageScenario
import com.example.performance.config.TestConfig

class SmokeSimulation extends Simulation {

  val httpProtocol = http
    .baseUrl(TestConfig.baseUrl)
    .acceptHeader("text/html")
    .userAgentHeader("Mozilla/5.0")

  setUp(
    HomepageScenario.homepage.inject(atOnceUsers(5))
  ).protocols(httpProtocol)
   .assertions(
     global.successfulRequests.percent.gt(95),
     global.responseTime.max.lt(3000)
   )
}

File: src/test/java/com/example/performance/simulations/SmokeSimulation.java
package com.example.performance.simulations;

import com.example.performance.config.TestConfig;
import com.example.performance.scenarios.HomepageScenario;

import static io.gatling.javaapi.core.CoreDsl.atOnceUsers;
import static io.gatling.javaapi.core.CoreDsl.global;
import static io.gatling.javaapi.core.CoreDsl.scenario;
import io.gatling.javaapi.core.Simulation;
import static io.gatling.javaapi.http.HttpDsl.http;
import io.gatling.javaapi.http.HttpProtocolBuilder;

public class SmokeSimulation extends Simulation {

    HttpProtocolBuilder httpProtocol = http
            .baseUrl(TestConfig.BASE_URL)
            .acceptHeader("text/html")
            .userAgentHeader("Mozilla/5.0");

    {
        setUp(
                scenario("Smoke Test").exec(HomepageScenario.homepage).injectOpen(
                        atOnceUsers(5)
                )
        ).protocols(httpProtocol)
                .assertions(
                        global().successfulRequests().percent().gt(95.0),
                        global().responseTime().max().lt(Integer.valueOf(3000))
                );
    }
}